testhelpers/image_privileges.go

```diff
@@ -0,0 +1,35 @@
+package testhelpers
+
+import (
+	"regexp"
+	"strings"
+)
+
+var urlRegex = regexp.MustCompile(`v2\/(.*)\/(?:blobs|manifests|tags)`)
+
+type ImagePrivileges struct {
+	readable  bool
+	writeable bool
+}
+
+func NewImagePrivileges(imageName string) *ImagePrivileges {
+	var isReadable, isWriteable bool
+	if strings.Contains(imageName, "readable") {
+		isReadable = true
+	}
+	if strings.Contains(imageName, "writable") {
+		isWriteable = true
+	}
+	return &ImagePrivileges{readable: isReadable, writeable: isWriteable}
+}
```

Comment:
```suggestion
func NewImagePrivileges(imageName string) (priv ImagePrivileges) {
	priv.readable = strings.Contains(imageName, "readable")
	priv.writeable = strings.Contains(imageName, "writable")
	return
}
```

I liked the old interface better. But you're using it, not me 😅

Reply:
I don't see who is using this function but I think the `if` conditions here are redundant.
I'm ok with returning a pointer, but I would remove the conditions and just use the return value from `strings.Contains` to set the variables.

Reply:
I added this way of setting the properties based on @natalieparellano comments [here](https://github.com/buildpacks/lifecycle/pull/646#discussion_r669816609), I will use this method in the lifecycle unit tests

Reply:
Fixed in the following [commit](https://github.com/buildpacks/imgutil/pull/125/commits/b6d4eb2c54b191a59bf4a46c43c492a881e6889b)